types/src/jwks/mod.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    jwks::{rsa::RSA_JWK, unsupported::UnsupportedJWK},
+    move_any::{Any as MoveAny, AsMoveAny},
+};
+use aptos_crypto::bls12381;
+use aptos_crypto_derive::{BCSCryptoHash, CryptoHasher};
+use move_core_types::account_address::AccountAddress;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeSet;
+
+pub type Issuer = Vec<u8>;
+
+pub fn issuer_from_str(s: &str) -> Issuer {
+    s.as_bytes().to_vec()
+}
+
+/// Move type `0x1::jwks::OIDCProvider` in rust.
+pub struct OIDCProvider {
+    pub name: Issuer,
+    pub config_url: Vec<u8>,
+}
+
+/// Move type `0x1::jwks::SupportedOIDCProviders` in rust.
+pub struct SupportedOIDCProviders {
+    pub providers: Vec<OIDCProvider>,
+}
+
+/// Reflection of Move type `0x1::jwks::JWK`.
+/// When you load an on-chain config that contains some JWK(s), the JWK will be of this type.
+/// When you call a Move function from rust that takes some JWKs as input, pass in JWKs of this type.
+/// Otherwise, it is recommended to convert this to the rust enum `JWK` for better rust experience.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize, CryptoHasher, BCSCryptoHash)]
+pub struct JWKMoveStruct {
```

Comment:
Since all of the other structs follow the same naming as in `jwks.move`, don't you wanna be consistent here and use `JWK` and rename the enum below to `JwkType` or something like that?

Reply:
I expect other rust components to reference `JWK` much more than `JWKMoveStruct`, so giving it a better name.

both structs comments are updated and should clarify enough?

Reply:
🤷